CalibCalorimetry/HcalTPGAlgos/src/HcaluLUTTPGCoder.cc

```diff
@@ -275,210 +285,226 @@ double HcaluLUTTPGCoder::cosh_ieta(int ieta, int depth, HcalSubdetector subdet)
+      }
+      ped /= 4.0;
+      gain /= 4.0;
+
+      //Get Channel Quality
+      const HcalChannelStatus* channelStatus = conditions.getHcalChannelStatus(cell);
+      status = channelStatus->getValue();
+
+    } else {
+      const HcalL1TriggerObject* myL1TObj = conditions.getHcalL1TriggerObject(cell);
+      ped = myL1TObj->getPedestal();
+      gain = myL1TObj->getRespGain();
+      status = myL1TObj->getFlag();
+    }  // LUTGenerationMode_
+
+    ped_[lutId] = ped;
+    gain_[lutId] = gain;
+    bool isMasked = ((status & bitToMask_) > 0);
+    float rcalib = meta->getRCalib();
+
+    auto adc2fC = [channelCoder, shape](unsigned int adc) {
+      float fC = 0;
+      for (auto capId : {0, 1, 2, 3})
+        fC += channelCoder->charge(*shape, adc, capId);
+      return fC / 4;
+    };
+
+    int qieType = conditions.getHcalQIEType(cell)->getValue();
+
+    const size_t SIZE = qieType == QIE8 ? INPUT_LUT_SIZE : UPGRADE_LUT_SIZE;
+    const int MASK = qieType == QIE8 ? QIE8_LUT_BITMASK : qieType == QIE10 ? QIE10_LUT_BITMASK : QIE11_LUT_BITMASK;
+    double linearLSB = linearLSB_QIE8_;
+    if (qieType == QIE11 and cell.ietaAbs() == topo_->lastHBRing())
+      linearLSB = linearLSB_QIE11Overlap_;
+    else if (qieType == QIE11)
+      linearLSB = linearLSB_QIE11_;
+
+    lut.resize(SIZE, 0);
+
+    // Input LUT for HB/HE/HF
+    if (subdet == HcalBarrel || subdet == HcalEndcap) {
+      int granularity = meta->getLutGranularity();
+
+      double correctionPhaseNS = conditions.getHcalRecoParam(cell)->correctionPhaseNS();
+      for (unsigned int adc = 0; adc < SIZE; ++adc) {
+        if (isMasked)
+          lut[adc] = 0;
+        else {
+          double nonlinearityCorrection = 1.0;
+          double containmentCorrection2TSCorrected = 1.0;
+          // SiPM nonlinearity was not corrected in 2017
+          // and containment corrections  were not
+          // ET-dependent prior to 2018
+          if (is2018OrLater) {
+            double containmentCorrection1TS = pulseCorr_->correction(cell, 1, correctionPhaseNS, adc2fC(adc));
+            // Use the 1-TS containment correction to estimate the charge of the pulse
+            // from the individual samples
+            double correctedCharge = containmentCorrection1TS * adc2fC(adc);
+            containmentCorrection2TSCorrected = pulseCorr_->correction(cell, 2, correctionPhaseNS, correctedCharge);
+            if (qieType == QIE11) {
+              const HcalSiPMParameter& siPMParameter(*conditions.getHcalSiPMParameter(cell));
+              HcalSiPMnonlinearity corr(
+                  conditions.getHcalSiPMCharacteristics()->getNonLinearities(siPMParameter.getType()));
+              const double fcByPE = siPMParameter.getFCByPE();
+              const double effectivePixelsFired = correctedCharge / fcByPE;
+              nonlinearityCorrection = corr.getRecoCorrectionFactor(effectivePixelsFired);
+            }
+          }
+          if (allLinear_)
```

Comment:
@smuzaffar the following splitting does not look nice, but I am afraid there not much that can be done in this case